zerver/tests/test_users.py

```diff
@@ -265,7 +265,7 @@ def test_admin_user_can_change_full_name(self) -> None:
         self.assert_json_success(result)
         hamlet = self.example_user('hamlet')
         self.assertEqual(hamlet.full_name, new_name)
-        req['is_admin'] = ujson.dumps(False)
+        req['role'] = ujson.dumps(UserProfile.ROLE_MEMBER)
         result = self.client_patch('/json/users/{}'.format(hamlet.id), req)
         self.assert_json_success(result)
```

Comment:
Can you extend this test to verify that the user's role was indeed changed?  I'm generally very suspicious of tests that end with `assert_json_success` but don't check that anything actually happened afterwards; that's what allows noop test bugs like this to creep in.

Reply:
I actually not thought of this initially, but I am not able to understand why are we changing role in this test which checks for change of name of user and hamlet is anyway already a member.

Reply:
That is a good point.  OK, let's instead just have a commit that deletes the last 4 lines of this test as being useless, and a second that just fixes the other test?

Reply:
Oh, 41fbb16cdfdb737ed3f57428ce2720d332ef68a1 is what introduced those lines; I think our change of the `role` API may have fixed the need to test that?

Reply:
Actually, I do not understand the need of adding those lines at that time also, it just checks to remove admin after changing name and that too from a user who is not admin.
Certainly, there is no need of those lines.

Reply:
The actual change in non-test code, was to ensure that we do not raise error if we pass `is_admin=false` to already non-admin user as we are not actuallly removing admin status of any user, it was no way related to change in name or any other property change.

Reply:
@timabbott I have added separate commit for removing redundant lines from the test.